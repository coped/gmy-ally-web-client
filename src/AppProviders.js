import React from "react";
import { AuthProvider } from "context/auth";
import { UserProvider } from "context/user";
import { ExercisesProvider } from "context/exercises";

export default function AppProviders({ children }) {
  return (
    <AuthProvider>
      <UserProvider>
        <ExercisesProvider>{children}</ExercisesProvider>
      </UserProvider>
    </AuthProvider>
  );
}
