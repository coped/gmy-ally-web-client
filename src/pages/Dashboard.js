import React from "react";

export default function Dashboard() {
  return <p>This is the dashboard.</p>;
}
