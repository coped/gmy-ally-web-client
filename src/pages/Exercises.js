import React, { useEffect, useState } from "react";
import Api from "lib/api";
import { PageLoader } from "components/common";
import { useExercises } from "context/exercises";

export default function Exercises() {
  const {
    allExercises,
    muscleGroups,
    categorizedExercises,
    setAllExercises,
  } = useExercises();
  const [loading, setLoading] = useState(false);
  const [currentMuscleGroup, setCurrentMuscleGroup] = useState("abdominals");

  // Fetch exercise data on component mount
  useEffect(() => {
    async function fetchData() {
      setLoading(true);
      const data = await Api.indexExercises();
      setAllExercises(data.payload.exercises);
      setLoading(false);
    }
    !allExercises && fetchData();
  }, []);

  function showThisMuscleGroup(name) {
    setCurrentMuscleGroup(name);
  }

  function showExercises() {
    console.log(allExercises);
    allExercises && console.log(categorizedExercises);
  }

  function capitalize(string) {
    return string[0].toUpperCase() + string.substring(1);
  }

  return (
    <div id="Exercises">
      <PageLoader loading={loading} className="is-info" />
      <div className="columns">
        <div className="column is-3">
          <aside className="menu">
            <p className="menu-label">SEARCH EXERCISES BY MUSCLE GROUP</p>
            <ul className="menu-list">
              {muscleGroups &&
                muscleGroups.map((name) => (
                  <li key={name}>
                    <a onClick={() => showThisMuscleGroup(name)}>
                      {capitalize(name)}
                    </a>
                  </li>
                ))}
            </ul>
          </aside>
        </div>
        <div className="column">
          <h3 className="is-size-3 has-text-centered">
            {capitalize(currentMuscleGroup)}
          </h3>
          <div className="box">
            <ul className="menu-list">
              {categorizedExercises &&
                categorizedExercises[currentMuscleGroup].map((exercise) => (
                  <li className="list-item">
                    <a>{exercise.title}</a>
                  </li>
                ))}
            </ul>
          </div>
        </div>
      </div>
      <button onClick={showExercises} className="button">
        Show exercises
      </button>
    </div>
  );
}
