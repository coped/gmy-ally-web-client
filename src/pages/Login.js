import React, { useState } from "react";
import { Link, Redirect } from "react-router-dom";
import { FormInput, FormButton } from "components/form";
import { Notification } from "components/common";
import { useAuth } from "context/auth";
import { useUser } from "context/user";
import Api from "lib/api";
import "assets/Login.scss";

export default function Login() {
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const [apiMessages, setApiMessages] = useState([]);
  const [form, setForm] = useState({ email: "", password: "" });

  const { setAuthContext } = useAuth();
  const { setUserContext } = useUser();

  function onChange(event) {
    const target = event.target;
    setForm({
      ...form,
      [target.name]: target.value,
    });
  }

  async function loginUser(e) {
    e.preventDefault();
    setIsLoading(true);
    const data = await Api.login({ info: form });
    if (data.status === "success") {
      setAuthContext(data.payload.jwt);
      setUserContext(data.payload.user);
      setIsLoggedIn(true);
    } else {
      setApiMessages(data.messages);
      setIsLoading(false);
    }
  }

  if (isLoggedIn) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <div id="Login">
      <h1 className="login-title">Log in</h1>
      <form onSubmit={loginUser} className="column is-6 box">
        {apiMessages &&
          apiMessages.map((response, index) => (
            <Notification key={index} classList={[`is-${response.type}`]}>
              <p>{response.message}</p>
            </Notification>
          ))}
        <FormInput
          id="email-input"
          label="Email:"
          name="email"
          type="email"
          placeholder="your@email.com"
          value={form.email}
          onChange={onChange}
        />
        <FormInput
          id="password-input"
          label="Password:"
          name="password"
          type="password"
          value={form.password}
          onChange={onChange}
        />
        <FormButton classList={["is-link"]} loading={isLoading}>
          Log in
        </FormButton>
        <p>
          Don't have an account? <Link to="/signup">Sign up</Link>.
        </p>
      </form>
    </div>
  );
}
