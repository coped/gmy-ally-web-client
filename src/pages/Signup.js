import React, { useState } from "react";
import { Notification } from "components/common";
import { FormInput, FormButton } from "components/form";
import { Link, Redirect } from "react-router-dom";
import Api from "lib/api";
import { useAuth } from "context/auth";
import "assets/Signup.scss";

export default function Signup(props) {
  const [isLoading, setIsLoading] = useState(false);
  const [form, setForm] = useState({
    name: "",
    email: "",
    password: "",
    password_confirmation: "",
  });
  const [apiMessages, setApiMessages] = useState("");
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const { setAuthToken } = useAuth();

  async function signupUser(e) {
    e.preventDefault();
    setIsLoading(true);
    const data = await Api.createUser(form);
    if (data.status === "success") {
      setAuthToken(data.payload.jwt);
      setIsLoggedIn(true);
    } else {
      setApiMessages(data.messages);
      setIsLoading(false);
    }
  }

  function onChange(event) {
    const target = event.target;
    setForm({ ...form, [target.name]: target.value });
  }

  if (isLoggedIn) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <div id="Signup">
      <h1 className="signup-title">Sign up</h1>
      <form onSubmit={signupUser} className="column is-6 box">
        {apiMessages &&
          apiMessages.map((message, index) => (
            <Notification key={index} type={message.type}>
              <p>{message.message}</p>
            </Notification>
          ))}
        <FormInput
          id="name-input"
          label="Name:"
          name="name"
          type="text"
          placeholder="Your name"
          value={form.name}
          onChange={onChange}
        />
        <FormInput
          id="email-input"
          label="Email:"
          name="email"
          type="email"
          placeholder="your@email.com"
          value={form.email}
          onChange={onChange}
        />
        <FormInput
          id="password-input"
          label="Password:"
          name="password"
          type="password"
          value={form.password}
          onChange={onChange}
        />
        <FormInput
          id="password-confirmation-input"
          label="Password confirmation:"
          name="passwordConfirmation"
          type="password"
          value={form.passwordConfirmation}
          onChange={onChange}
        />
        <FormButton classList={["is-link"]} loading={isLoading}>
          Sign up
        </FormButton>
        <p className="is-size-6">
          Already have an account? <Link to="/login">Log in</Link>.
        </p>
      </form>
    </div>
  );
}
