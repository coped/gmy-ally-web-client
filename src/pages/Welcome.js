import React from "react";
import "assets/Welcome.scss";
import "react-responsive-carousel/lib/styles/carousel.min.css";
import { Carousel } from "react-responsive-carousel";
import Larry from "images/larry-1x1.jpg";
import Sandy from "images/buff-sandy-1x1.jpg";
import EyeMuscles from "images/eyeball-muscles-1x1.jpg";
import MusclesOnMuscles from "images/muscles-on-muscles-1x1.jpg";
import KaliMuscle from "images/kali-muscle-1x1.jpg";
import BelgianBlue from "images/belgian-blue-1x1.jpg";
import { Link } from "react-router-dom";

export default function Welcome() {
  return (
    <div id="Welcome">
      <h1 className="title has-text-centered is-size-1">
        Welcome to Gym Partner!
      </h1>
      <div className="columns section">
        <div className="column content">
          <div className="container">
            <h2 className="has-text-centered">
              Gym Partner is a web and mobile application built to help you:
            </h2>
            <div>
              <ul className="is-size-5">
                <li>Schedule workout routines</li>
                <li>Set fitness goals</li>
                <li>Track your improvement over time</li>
              </ul>
            </div>
          </div>
          <div className="has-text-centered section">
            <h2>Sign up and try it out!</h2>
            <Link to="/signup" className="button is-primary is-large">
              <strong>Sign up</strong>
            </Link>
          </div>
        </div>
        <div className="column is-5">
          <div className="box">
            <Carousel
              showArrows={false}
              autoPlay={true}
              showThumbs={false}
              showStatus={false}
              showIndicators={false}
              infiniteLoop={true}
              interval={2500}
              dynamicHeight={true}
              transitionTime={500}
              stopOnHover={false}
            >
              <div>
                <img src={Larry} alt="Larry the Lobster" />
              </div>
              <div>
                <img src={KaliMuscle} alt="Kali muscle" />
              </div>
              <div>
                <img src={Sandy} alt="Sandy Cheeks" />
              </div>
              <div>
                <img src={BelgianBlue} alt="Belgian blue" />
              </div>
              <div>
                <img src={EyeMuscles} alt="Eyeball muscles" />
              </div>
              <div>
                <img src={MusclesOnMuscles} alt="Muscles on muscles" />
              </div>
            </Carousel>
          </div>
        </div>
      </div>
    </div>
  );
}
