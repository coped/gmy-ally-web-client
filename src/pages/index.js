import Login from "pages/Login";
import Signup from "pages/Signup";
import Welcome from "pages/Welcome";
import Dashboard from "pages/Dashboard";
import Exercises from "pages/Exercises";

export { Login, Signup, Welcome, Dashboard, Exercises };
