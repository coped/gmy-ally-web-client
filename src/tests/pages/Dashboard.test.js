import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Dashboard as DashboardComponent } from "pages";
import { MemoryRouter } from "react-router-dom";
import AppProviders from "AppProviders";

Enzyme.configure({ adapter: new Adapter() });

const Dashboard = (
  <AppProviders>
    <MemoryRouter initialEntries={["/dashboard"]}>
      {<DashboardComponent />}
    </MemoryRouter>
  </AppProviders>
);

describe("Dashboard", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(Dashboard, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(Dashboard);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
