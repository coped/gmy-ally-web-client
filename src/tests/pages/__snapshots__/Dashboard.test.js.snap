// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard has a valid snapshot 1`] = `
<div
  id="Dashboard"
>
  <div
    className="pageloader is-info"
  >
    <span
      className="title"
    >
      Loading . . .
    </span>
  </div>
  <p>
    This is the dashboard.
  </p>
  <button
    className="button"
    onClick={[Function]}
  >
    Show data
  </button>
</div>
`;
